### PR TITLE
Add authorized recipe controller

### PR DESCRIPTION
**Pull request by mplekunov**
Depends on #94

### CODE REVIEW

**Comment by mplekunov** [comment]
The recipe routes have been rewritten to fit our new specifications:
Routes differentiate between two cases: Unauthorized && Authorized users.
 
Unauthorized case happens when request doesn't include "Authorization" header.
Authorized case happens when request includes "Authorization" header.

When user is Authorized. Responses (in both get/getAll calls) will contain additional field "isFavorite". When it's set to true, it means this recipe is part of the favorite recipes list of the user and vice versa.

**Comment by nashirj** [comment]
Pls fix merge conflicts

File: server/serverAPI/controller/AuthorizedRecipeController.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+export default class RecipeController extends BaseUserController {
+    private recipeAPI: IRecipeAPI;
+
+    constructor(database: IDatabase<IUser>, recipeAPI: IRecipeAPI) {
+        super(database);
+
+        this.recipeAPI = recipeAPI;
+    }
+
+    protected async convertToUserRecipe(recipe: IRecipe, req: Request, res: Response): Promise<UserRecipe> {
```
Comment (nashirj) [inline server/serverAPI/controller/AuthorizedRecipeController.ts:25]: Can you update this and below to have logic for allergens as well please

File: server/serverAPI/model/recipe/UserBaseRecipe.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import IImage from "../image/IImage";
+import IBaseRecipe from "./IBaseRecipe";
+
+export default class UserBaseRecipe implements IBaseRecipe {
+    id: number;
+    name: string;
+    image: IImage;
+    isFavorite: boolean;
+
+    constructor(recipe: IBaseRecipe, isFavorite: boolean) {
```
Comment (nashirj) [inline server/serverAPI/model/recipe/UserBaseRecipe.ts:10]: Please add `hasAllergen`

**Review by nashirj — CHANGES_REQUESTED** [review]

**Comment by nashirj** [comment]
Overall looks good, just pls add logic for hasAllergen

**Pull request closed without merging**